android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -647,6 +650,56 @@ class GeckoEngineSession(
         )
     }
 
+    /**
+     * See [EngineSession.requestProductAnalysis]
+     */
+    override fun requestProductAnalysis(
+        url: String,
+        onResult: (ShoppingProductAnalysis) -> Unit,
+        onException: (Throwable) -> Unit,
+    ) {
+        geckoSession.requestAnalysis(url).then({
+                response ->
+            if (response == null) {
+                logger.error(
+                    "Invalid value: unable to get analysis result from Gecko Engine.",
+                )
+                onException(
+                    java.lang.IllegalStateException(
+                        "Invalid value: unable to get analysis result from Gecko Engine.",
+                    ),
+                )
+                return@then GeckoResult()
+            }
+            val highlights = Highlight(
```

Comment:
It would be nice to have a check if all the highlights are null, then this `highlights` is set to null and not 
Highlight(null, null, null, null, null).

Reply:
Is there a nice idiomatic Kotlin way of doing that? When I do the simple thing (check properties sequentially), the linter is not happy.

Reply:
As we discussed in Slack, I will make this a simple `if else` and check every prop sequentially, and suppress the lint warning.